packages/beacon-node/src/chain/blocks/types.ts

```diff
@@ -32,20 +34,30 @@ export enum GossipedInputType {
 }
 
 export type BlobsCache = Map<number, {blobSidecar: deneb.BlobSidecar; blobBytes: Uint8Array | null}>;
+
+type ForkBlobsInfo = {fork: ForkName.deneb};
+type BlobsData = {blobs: deneb.BlobSidecars; blobsBytes: (Uint8Array | null)[]; blobsSource: BlobsSource};
+export type BlockInputDataBlobs = ForkBlobsInfo & BlobsData;
+export type BlockInputData = BlockInputDataBlobs;
+
+type BlobsInputCache = {blobsCache: BlobsCache};
+export type BlockInputCacheBlobs = ForkBlobsInfo & BlobsInputCache;
+
 export type BlockInputBlobs = {blobs: deneb.BlobSidecars; blobsBytes: (Uint8Array | null)[]; blobsSource: BlobsSource};
-type CachedBlobs = {
-  blobsCache: BlobsCache;
-  availabilityPromise: Promise<BlockInputBlobs>;
-  resolveAvailability: (blobs: BlockInputBlobs) => void;
-};
+type Availability<T> = {availabilityPromise: Promise<T>; resolveAvailability: (data: T) => void};
+
+type CachedBlobs = BlobsInputCache & Availability<BlockInputDataBlobs>;
```

Comment:
having both `CachedBlobs` and `BlobsCache` types seems confusing

Reply:
cleaned it up a bit